src/types/api.js

```diff
@@ -73,6 +73,34 @@
  * @prop {boolean} [$anchorTimeout]
  */
 
+/**
+ * @typedef AnnotationData
+ * @prop {string} uri
+ * @prop {string} $tag
+ * @prop {string} [$highlight]
+ * @prop {SelectSource[]} target
+ * @prop {Document} document
+ */
+
+/**
+ * @typedef SelectSource
+ * @prop {string} source
+ * @prop {Selector[]} selector
+ */
+
+/**
+ * @typedef Document
+ * @prop {string} title
+ * @prop {string[]} link
```

Comment:
`link` is an array of objects with `href` and `ref` properties rather than strings. See https://github.com/hypothesis/client/blob/c2ca6c8f6c3afbfddfacc0ebcf158ae610a1ea2f/src/annotator/plugin/document.js#L189. Note that the `ref` and `type` properties on that line are only currently written/read on the "annotator" side of the application.

Reply:
good catch! I'm not sur eif they are required or not, but I lean on making everything required until I see errors.